# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmInstance.cpp
# Commit: 5718a8d121f9
# Full Hash: 5718a8d121f95ce740a11ee443444cea990d6323
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 7: Remove Frame::tls. r=lth
#   
#   We remove Frame::tls and fix prologue/epilogue. Runtime, Ion and Baseline are ready for this move already.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83064
# ==============================================================================

diff -r 40031ad88dbd -r 5718a8d121f9 js/src/wasm/WasmInstance.cpp
--- a/js/src/wasm/WasmInstance.cpp	Tue Nov 03 10:28:10 2020 +0000
+++ b/js/src/wasm/WasmInstance.cpp	Tue Nov 03 10:32:55 2020 +0000
@@ -1752,8 +1752,17 @@
   // This is so as to ensure there are no areas of stack inadvertently ignored
   // by a stackmap, nor covered by two stackmaps.  Hence any failure of this
   // assertion is serious and should be investigated.
+
+  // This condition isn't kept for Cranelift
+  // (https://github.com/bytecodealliance/wasmtime/issues/2281), but this is ok
+  // to disable this assertion because when CL compiles a function, in the
+  // prologue, it (generates code) copies all of the in-memory arguments into
+  // registers. So, because of that, none of the in-memory argument words are
+  // actually live.
+#ifndef JS_CODEGEN_ARM64
   MOZ_ASSERT_IF(highestByteVisitedInPrevFrame != 0,
                 highestByteVisitedInPrevFrame + 1 == scanStart);
+#endif
 
   uintptr_t* stackWords = (uintptr_t*)scanStart;
 